server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java

```diff
@@ -359,13 +371,54 @@ private SearchWsResponse doHandle(SearchRequest request, Request wsRequest) {
     // Must be done after loading of data as the "hidden" facet "debt"
     // can be used to get total debt.
     facets = reorderFacets(facets, options.getFacets());
+    replaceRuleIdsByRuleKeys(facets, data.getRules() == null ? emptyList() : data.getRules());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: "replaceRuleIdsByRuleKeys" is marked "javax.annotation.ParametersAreNonnullByDefault" but parameter 2 could be null. ([squid:S2637](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2637))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Feh%2FSONAR-10303&issues=AWFBQzvZmckxG3vItnLF&open=AWFBQzvZmckxG3vItnLF)

[ISSUE_KEY]: AWFBQzvZmckxG3vItnLF

Reply:
This issue has been resolved by a user in SonarQube

[ISSUE_STATUS]: RESOLVED